theano/gpuarray/dnn.py

```diff
@@ -1860,6 +1860,8 @@ def local_gpua_pool_dnn_alternative(op, ctx_name, inputs, outputs):
     if not op.ignore_border:
         return
     img, = inputs
+    if len(op.ds) not in (2, 3) or img.ndim != len(op.ds) + 2:
```

Comment:
Your explanation is right. I made this comment before I understand that you extended what the op does to support more dimensions.